### PR TITLE
#4507: handle different user-selected tags when inferring expanded selection

### LINKED ISSUE #4507: Check for common tag type when generating selector for expanded selection
Context
---
- Currently the commonCssSelector method uses the first tag type in generate selectors
- commonCssSelector should never return a selector that doesn't match all elements passed to it
- https://www.notion.so/pixiebrix/Todd-Jay-Selector-Inference-2022-10-19-72608603df0b4469857244d75d18b2b0

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Closes #4507
- Rework of https://github.com/pixiebrix/pixiebrix-extension/pull/4535

## Reviewer Tips

- There will be additional follow-up work on the heuristics. This PR just gets rid of the buggy behavior w.r.t. tags

## Future Work

- Support other non-unique attributes: https://github.com/pixiebrix/pixiebrix-extension/issues/4508

## Checklist

- [X] Add tests
- [X] Run Storybook and manually confirm that all stories are working
- [X] Designate a primary reviewer: @BLoe (to get more exposure to inference logic)

### CODE REVIEW

File: src/contentScript/nativeEditor/selectorInference.test.ts [hunk 1]
```diff
@@ -256,41 +256,50 @@ describe("safeCssSelector", () => {
   /* eslint-enable jest/expect-expect */
 });
 
-describe("commonCssSelector", () => {
+describe("expandedCssSelector", () => {
   /* eslint-disable jest/expect-expect -- Custom expectSelector */
-  const expectSelector = (selector: string, body: string) => {
+  const expectRountripSelector = (selector: string, body: string) => {
```
Comment (twschiller) [inline src/contentScript/nativeEditor/selectorInference.test.ts:261]: Clarify how this expect work. It expects the exact same elements as provided, so has relatively little use for the expansion use case

File: src/contentScript/nativeEditor/selectorInference.test.ts [hunk 2]
```diff
@@ -256,41 +256,50 @@ describe("safeCssSelector", () => {
...
-    ];
-    const commonSelector = commonCssSelector(elements);
+    const elements = [
+      ...document.body.querySelectorAll(selector),
+    ] as HTMLElement[];
+    const commonSelector = expandedCssSelector(elements);
     expect(commonSelector).toBe(selector);
   };
 
+  const expectSelector = (selector: string, expected: string, body: string) => {
```
Comment (twschiller) [inline src/contentScript/nativeEditor/selectorInference.test.ts:270]: The more general Jest expect

File: src/contentScript/nativeEditor/selectorInference.test.ts [hunk 3]
```diff
@@ -256,41 +256,50 @@ describe("safeCssSelector", () => {
...
+    document.body.innerHTML = body;
+    const elements = [
+      ...document.body.querySelectorAll(selector),
+    ] as HTMLElement[];
+    const commonSelector = expandedCssSelector(elements);
+    expect(commonSelector).toBe(expected);
+  };
+
   const expectSimilarElements = (
```
Comment (twschiller) [inline src/contentScript/nativeEditor/selectorInference.test.ts:279]: Generalized to work with any number of elements

File: src/contentScript/nativeEditor/selectorInference.ts [hunk 4]
```diff
@@ -386,59 +409,69 @@ export function commonCssSelector(
...
+    root,
+  });
+
+  // Get common attribute of parent of user selected elements. Will be used before to use the ">" immediate descendant
+  // operator for the elements.
+  //
+  // Using the parent or another level of ancestor is helpful for preventing over-matching on based on the user-selected
+  // elements attribute/tag name selector.
+  //
+  // TODO: look for all common attributes that could be non-unique, e.g., aria-role="button"
```
Comment (twschiller) [inline src/contentScript/nativeEditor/selectorInference.ts:437]: TODOs are here: https://github.com/pixiebrix/pixiebrix-extension/issues/4508

**Review by BLoe — APPROVED** [review]

**Pull request merged**